core/vm/evm.go

```diff
@@ -337,13 +346,14 @@ func (evm *EVM) CallCode(caller ContractRef, addr common.Address, input []byte,
 // DelegateCall differs from CallCode in the sense that it executes the given address'
 // code with the caller as context and the caller is set to the caller of the caller.
 func (evm *EVM) DelegateCall(caller ContractRef, addr common.Address, input []byte, gas uint64) (ret []byte, leftOverGas uint64, err error) {
+	caller = evm.maybeOverrideCaller(caller)
 	// Invoke tracer hooks that signal entering/exiting a call frame
 	if evm.Config.Tracer != nil {
 		// NOTE: caller must, at all times be a contract. It should never happen
 		// that caller is something other than a Contract.
-		parent := caller.(*Contract)
+		parent := caller.(interface{ Value() *uint256.Int })
 		// DELEGATECALL inherits value from parent call
-		evm.captureBegin(evm.depth, DELEGATECALL, caller.Address(), addr, input, gas, parent.value.ToBig())
+		evm.captureBegin(evm.depth, DELEGATECALL, caller.Address(), addr, input, gas, parent.Value().ToBig())
```

Comment:
Could the `maybeOverrideCaller` call handle things in a way that makes this interface casting not needed? Seems like we were previously confident that caller would have `Value()`, so after using the prank function I would hope it still can.

Reply:
The issue is that geth breaks the interface, and inspects if it's a contract. Whereas the override might substitute this caller. As override we don't want to re-initialize a full replacement `Contract` caller context, but instead want to just provide the thing that matters: a caller, with a value. Using an interface to retrieve the value from the `Contract`, with the already existing `Value` method, solves this issue.